docs/ui/ComponentDemo.tsx

```diff
@@ -1,28 +1,41 @@
-import { ReactNode } from 'react';
-import { registry } from '../registry';
-
+import { ReactNode, useState } from 'react';
+import { Button, MarigoldProvider } from './marigold';
+import { useThemeSwitch } from '@/app/components/[...slug]/_components';
+import { Theme } from '../../packages/system/src';
+import { registry } from '@/registry';
 export interface ComponentDemoProps {
-  name: keyof typeof registry;
-  source: string;
+  name: string;
   children?: ReactNode;
 }
 
-export const ComponentDemo = ({
-  name,
-  source,
-  children,
-}: ComponentDemoProps) => {
+export const ComponentDemo = ({ name, children }: ComponentDemoProps) => {
   const Demo = registry[name].demo;
-
+  const { current, themes } = useThemeSwitch();
+  const [showCode, setShowCode] = useState(true);
```

Comment:
Why not use the Marigold Tabs :D

Reply:
I agree this is the First option came to my mind 
but i wanted it to look like shadcn "more elegant"

Reply:
but you can style them like shadcn :)